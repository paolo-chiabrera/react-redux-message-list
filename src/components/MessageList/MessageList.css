.message-list {
    list-style: none;
}
.message-list .date {
    font-style: italic;
}