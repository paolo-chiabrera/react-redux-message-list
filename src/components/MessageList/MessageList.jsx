import React from 'react';

import './MessageList.css';

const MessageList = ({ messages = [] }) => {
    const _messages = messages.map(item => {
        const { member } = item;
        const initials = member.firstName[0].toUpperCase() + member.lastName[0].toUpperCase();

        return (
            <li key={item.id} className="message-item">
                <div className="row">
                    <div className="column column-20">
                        <img src={member.avatar} alt={initials}/>
                    </div>
                    <div className="column column-50">
                        <p title={member.email} >{item.message}</p>
                    </div>
                    <div className="column column-30">
                        <p className="date">{item.humanReadableTimestamp}</p>
                    </div>
                </div>
            </li>
        );
    });

    return (
        <ul className="message-list">
            {_messages}
        </ul>
    );
};

export default MessageList;