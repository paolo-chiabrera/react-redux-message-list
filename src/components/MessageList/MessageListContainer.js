import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { compose, lifecycle } from 'recompose';
import moment from 'moment';

import getMessages from '../../actions/getMessages';

import MessageList from './MessageList.jsx';

export const remapMessages = (messages = [], members = {}) => {
    return messages
        .map(message => {
            const M = moment(message.timestamp);

            return Object.assign({
                humanReadableTimestamp: M.format('Do MMMM YYYY, HH:mm'),
                unix: M.unix()
            }, message, {
                member: members[message.userId]
            });
        })
        .sort((prev, next) => next.unix - prev.unix);
};

export const mapStateToProps = state => {
    const { members, messages } = state;

    return {
        messages: remapMessages(messages, members)
    }
};

export const mapDispatchToProps = dispatch => bindActionCreators({ getMessages }, dispatch);

export const lifecycleMethods = {
    componentDidMount() {
        this.props.getMessages();
    }
};

export default compose(
    connect(
        mapStateToProps,
        mapDispatchToProps
    ),
    lifecycle(lifecycleMethods)
)(MessageList);