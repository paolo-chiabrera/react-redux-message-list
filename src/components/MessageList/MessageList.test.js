import React from 'react';
import renderer from 'react-test-renderer';

import MessageList from './MessageList.jsx';

test('Should render the MessageList component', () => {
    const component = renderer.create(<MessageList />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});