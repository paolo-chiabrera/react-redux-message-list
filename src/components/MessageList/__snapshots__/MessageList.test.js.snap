exports[`test Should render the MessageList component 1`] = `
<ul
  className="message-list" />
`;
